File: test-suite/src/test/java/io/micronaut/docs/function/aws/EventLogger.java
package io.micronaut.docs.function.aws;

//tag::clazz[]
import io.micronaut.function.FunctionBean;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import java.util.function.Consumer;

@FunctionBean("eventlogger")
public class EventLogger implements Consumer<String> {
    private static final Logger LOG = LoggerFactory.getLogger(EventLogger.class);
    @Override
    public void accept(String input) {
        LOG.info("Received: {}", input);
    }
}
//end::clazz[]

File: test-suite-kotlin/src/test/kotlin/io/micronaut/docs/function/aws/EventLogger.kt
package com.example
//tag::clazz[]
import io.micronaut.function.FunctionBean
import org.slf4j.LoggerFactory
import java.util.function.Consumer

@FunctionBean("eventlogger")
class EventLogger : Consumer<String> {
    override fun accept(input: String) {
        LOG.info("Received: {}", input)
    }

    companion object {
        private val LOG = LoggerFactory.getLogger(EventLogger::class.java)
    }
}
//end::clazz[]